File: lib/Model/profileModel.dart
import 'package:json_annotation/json_annotation.dart';

part 'profileModel.g.dart';

@JsonSerializable()
class ProfileModel {
  String? id;
  String username;
  String fullname;
  String DOB;
  String professor;
  String influencer;
  String bio;
  String img;
  String pinned;
  List listfollower;
  List listfollowing;
  List favorite;
  List notification;
  List saveproduct;

  ProfileModel({
    this.id,
    required this.username,
    required this.fullname,
    required this.DOB,
    required this.professor,
    required this.influencer,
    required this.bio,
    required this.img,
    required this.pinned,
    required this.listfollower,
    required this.listfollowing,
    required this.favorite,
    required this.notification,
    required this.saveproduct,
  });

  factory ProfileModel.fromJson(Map<String, dynamic> json) =>
      _$ProfileModelFromJson(json);

  Map<String, dynamic> toJson() => _$ProfileModelToJson(this);
}

File: Backend-Firebase/functions/models/profile.model.js
const mongoose = require("mongoose");

const schema = mongoose.Schema;

const Profile = schema(
    {
      username: {
        type: String,
        required: true,
        unique: true,
      },
      fullname: String,
      DOB: String,
      professor: {
        type: String,
        default: "",
      },
      influencer: {
        type: String,
        default: "",
      },
      bio: {
        type: String,
        default: "",
      },
      img: {
        type: String,
        default: "",
      },
      favorite: [
        {
          type: mongoose.Schema.Types.ObjectId,
          ref: "Post",
        },
      ],
      listfollower: [
        {
          img: {
            type: String,
            default: "",
          },
          fullname: {
            type: String,
          },
          username: {
            type: String,
            unique: true,
          },
        },
      ],
      listfollowing: [
        {
          img: {
            type: String,
            default: "",
          },
          fullname: {
            type: String,
          },
          username: {
            type: String,
            unique: true,
          },
        },
      ],
      saveproduct: [
        {
          type: mongoose.Schema.Types.ObjectId,
          ref: "Product",
        }
      ],
      pinned: {
        type: mongoose.Schema.Types.ObjectId,
        ref: "Pinned",
      },
      notification: [
        {
          type: mongoose.Schema.Types.ObjectId,
          ref: "notification",
        }
      ],
    },

    {
      timestamps: true,
    }
);

module.exports = mongoose.model("Profile", Profile);
